packages/modal/src/useModal.ts

```diff
@@ -48,37 +48,33 @@ export function useModal(
   const [idPrefix] = useState(_id || seed(`modal_${PACKAGE_VERSION}`));
   const titleId = `${idPrefix}--title`;
   const contentId = `${idPrefix}--content`;
-  const isModalMousedDownRef = useRef(false);
 
   const closeModal = (event: KeyboardEvent | MouseEvent) => {
     onClose && onClose(event);
   };
 
-  const getBackdropProps = ({ onMouseUp, ...other } = {} as any) => {
+  const getBackdropProps = ({ onClick, ...other } = {} as any) => {
     return {
-      onMouseUp: composeEventHandlers(onMouseUp, (event: MouseEvent) => {
-        if (!isModalMousedDownRef.current) {
+      onClick: composeEventHandlers(onClick, (event: MouseEvent) => {
+        const target = event.target as Element;
+
+        if (target?.getAttribute('data-garden-container-id') === 'containers.modal') {
           closeModal(event);
         }
-
-        isModalMousedDownRef.current = false;
```

Comment:
This `onMouseUp` was necessary to handle when users mouseDown within a modal, drag the cursor, then mouseUp outside modal. Otherwise we run into the following issue:

https://user-images.githubusercontent.com/4030377/111651346-60f56700-87c3-11eb-863e-6f9f9134face.mov

Reply:
With my change you need to get a click, so both the mouse down and mouse up needs to hit the backdrop.